source/common/tracing/opentracing_driver_impl.cc

```diff
@@ -0,0 +1,168 @@
+#include "common/tracing/opentracing_driver_impl.h"
+
+#include <sstream>
+
+#include "common/common/assert.h"
+#include "common/common/base64.h"
+
+namespace Envoy {
+namespace Tracing {
+
+namespace {
+class OpenTracingHTTPHeadersWriter : public opentracing::HTTPHeadersWriter {
+public:
+  explicit OpenTracingHTTPHeadersWriter(Http::HeaderMap& request_headers)
+      : request_headers_(request_headers) {}
+
+  // opentracing::HTTPHeadersWriter
+  opentracing::expected<void> Set(opentracing::string_view key,
+                                  opentracing::string_view value) const override {
+    request_headers_.addCopy(Http::LowerCaseString{key}, value);
+    return {};
+  }
+
+private:
+  Http::HeaderMap& request_headers_;
+};
+
+class OpenTracingHTTPHeadersReader : public opentracing::HTTPHeadersReader {
+public:
+  explicit OpenTracingHTTPHeadersReader(const Http::HeaderMap& request_headers)
+      : request_headers_(request_headers) {}
+
+  typedef std::function<opentracing::expected<void>(opentracing::string_view,
+                                                    opentracing::string_view)>
+      OpenTracingCb;
+
+  // opentracing::HTTPHeadersReader
+  opentracing::expected<opentracing::string_view>
+  LookupKey(opentracing::string_view key) const override {
+    const Http::HeaderEntry* entry;
+    Http::HeaderMap::Lookup lookup_result =
+        request_headers_.lookup(Http::LowerCaseString{key}, &entry);
+    switch (lookup_result) {
+    case Http::HeaderMap::Lookup::Found:
+      return opentracing::string_view{entry->value().c_str(), entry->value().size()};
+    case Http::HeaderMap::Lookup::NotFound:
+      return opentracing::make_unexpected(opentracing::key_not_found_error);
+    case Http::HeaderMap::Lookup::NotSupported:
+      return opentracing::make_unexpected(opentracing::lookup_key_not_supported_error);
+    default:
```

Comment:
default should not be required, you have all cases covered, compiler will check

Reply:
Without the default, I'll get this error when making the debug build:
```
source/common/tracing/opentracing_driver_impl.cc: In member function 'virtual opentracing::expected<opentracing::v1::string_view> Envoy::Tracing::{anonymous}::OpenTracingHTTPHeadersReader::LookupKey(opentracing::v1::string_view) const':
source/common/tracing/opentracing_driver_impl.cc:53:3: error: control reaches end of non-void function [-Werror=return-type]
   }
```

Reply:
just remove the default, and put the NOT_REACHED at the end of the function.

Reply:
Removed default case.